### ISSUE TITLE
Autofocus unreliable on Android

### ISSUE THREAD

**Issue by longlife1st** [capacitor-community/barcode-scanner#27]
**Describe the bug**
The camera sometimes does not autofocus properly. (It might be a problem happening on specific devices).
It is possible to fix the focus by switching to stock Camera app and return to app implementing this plugin.

**To Reproduce**
Implement QR reading using this plugin.

**Expected behavior**
Unless configured otherwise, plugin should have constant autofocus in order to find QR code easier.

**Screenshots**
<none>

**Version**
1.2.1

**Smartphone (please complete the following information):**

- Device: Xiaomi Mi10 Lite 5G
- OS: Android 10

**Additional context**
I was actually able to figure out what the problem is.
On my device, it seems that the focus setting is only working properly when the continuous mode is set.
Looking at the implementation of the plugin, I was actually able to fix the issue by adding the following code to the plugin, in the setupCamera() method.
```kotlin
settings.setContinuousFocusEnabled(true);
```
I was wondering if this mode was not added on purpose (maybe because it is not supported by all devices?) or if it just was not required at all so far.
At least, if it's not the case by default, I would like to add a way to add that flag through some kind of setting method.
I am open to writing a PR for that feature if necessary, but I'd like to hear if there was some design choice here I am not aware of.

**Comment by tafelnl** [comment]
This was not a specific design decision. I like the idea of being able to set a flag for this. Any PR for that is very welcome. Do you happen to know if something like this also exists on iOS?

**Comment by longlife1st** [comment]
@tafelnl 
First, sorry for the late reply, I haven't been able to take care of this until today.

I took a bit more time to look at how this is implemented in the zxing-android-embedded library.
Basically, even if we use 
```javascript
settings.setContinuousFocusEnabled(true);
```
the `CameraConfigurationUtils` class of the library actually selects to use Continuous Focus Mode only if it is one of the supported focus mode, as returned by Android OS.
[Implementation](https://github.com/journeyapps/zxing-android-embedded/blob/40260272fcff4f14181803495e7d370c23e35db7/zxing-android-embedded/src/com/journeyapps/barcodescanner/camera/CameraConfigurationUtils.java#L64)
Therefore, it doesn't seem to be too much of an issue to add that setting as default, as the chance it has adverse effect is pretty low.
Of course, having it as a flag is also definitely an option, but where to add it? As a `prepare()` option parameter would look like the best alternative...

> Do you happen to know if something like this also exists on iOS?

[It does seem to exist](https://developer.apple.com/documentation/avfoundation/avcapturedevice/focusmode/continuousautofocus)...
I have yet to test this thoroughly on the iOS version of the app I am working with, but there seems to be similar kind of things to check (check if continuous focus is supported, then activate it if it is...).

**Comment by tafelnl** [comment]
Closed by merging #29

Thanks @longlife1st
